src/components/Search.js

```diff
@@ -29,7 +29,7 @@ const debouncedSearch = debounce(
   (newValue, dispatch) => dispatch({ type: 'search', value: newValue })
 , SEARCH_DEBOUNCE_WAIT)
 
-export const Search = connect(({ search }) => ({ show: search != null }))(({ show, dispatch }) => {
+export const Search = connect(({ search }) => ({ show: search != null, search: search }))(({ show, search, dispatch }) => {
```

Comment:
If you are passing search through (which you can do simply with `({ search })`), I would recommend not passing `show` at all. In general, you only want to pass the minimum data across boundaries/interfaces that you need and then derive other values as late as possible.